File: solutions/cpp/raindrops/1/raindrops.cpp
#include "raindrops.h"
#include <string>

namespace raindrops {
std::string convert(int test) {
  std::string tmp{};
  if ((test % 3) == 0)
    tmp.append("Pling");
  if ((test % 5) == 0)
    tmp.append("Plang");
  if ((test % 7) == 0)
    tmp.append("Plong");
  if(tmp.empty())tmp=std::to_string(test);
  return tmp;
}
} // namespace raindrops

File: solutions/lua/raindrops/1/raindrops.lua
local function raindrops(n)
  local result = ""
  if (n % 3) == 0 then
    result = "Pling"
  end
  if (n % 5) == 0 then
    result = result .. "Plang"
  end
  if (n % 7) == 0 then
    result = result .. "Plong"
  end
  if #result == 0 then
    return tostring(n)
  end
  return result
end

return raindrops
